### PR TITLE
label y-axis for aggregation duration charts

### PR DESCRIPTION
**Pull request by michaelkirk**
Add y-axis labels to the aggregate duration chart.

**before**
Screen Shot 2021-07-13 at 8 53 20 PM

**after**

Screen Shot 2021-07-13 at 8 46 23 PM

(only the left chart changed)

### CODE REVIEW

File: geom/src/duration.rs [hunk 1]
```diff
@@ -81,7 +85,7 @@ impl Duration {
         remainder -= minutes * 60.0;
         let seconds = remainder.floor();
         remainder -= seconds;
-        let centis = (remainder / 0.1).floor();
+        let centis = (remainder / 0.01).round();
```
Comment (michaelkirk) [inline geom/src/duration.rs:88]: I think this was a bug - centi should be `seconds * 100` right?

File: geom/src/duration.rs [hunk 2]
```diff
@@ -196,6 +202,50 @@ impl Duration {
...
+    /// assert_eq!(Duration::seconds(3600.0).to_rounded_string(0), "1hr");
+    /// assert_eq!(Duration::seconds(3600.0).to_rounded_string(1), "1.0hr");
+    /// assert_eq!(Duration::seconds(7800.0).to_rounded_string(0), "2hr");
+    /// assert_eq!(Duration::seconds(800.0).to_rounded_string(1), "13.3min");
+    /// assert_eq!(Duration::seconds(-800.0).to_rounded_string(1), "-13.3min");
+    /// assert_eq!(Duration::seconds(0.0).to_rounded_string(0), "0");
+    /// assert_eq!(Duration::seconds(12.5).to_rounded_string(1), "12.5s");
+    /// assert_eq!(Duration::seconds(12.5).to_rounded_string(2), "12.50s");
+    /// ```
+    pub fn to_rounded_string(self, precision: usize) -> String {
```
Comment (michaelkirk) [inline geom/src/duration.rs:218]: I think that this kind of rounding works well for many of our tight spaces like axis labels.

File: game/src/sandbox/dashboards/travel_times.rs [hunk 3]
```diff
@@ -483,19 +493,60 @@ fn contingency_table(ctx: &mut EventCtx, app: &App, filter: &Filter) -> Widget {
...
+                intervals.1[1..].iter().map(|interval| {
+                    let y = total_height
+                        - max_bar_height
+                            * (1.0 - interval.abs().inner_seconds() / intervals.0.inner_seconds());
+                    (interval, y)
+                }),
+            )
+            .for_each(|(interval, y)| {
+                let start = Pt2D::new(0.0, y);
+                let line: geom::Line =
```
Comment (dabreegster) [inline game/src/sandbox/dashboards/travel_times.rs:529]: The type declaration isn't necessary, but tiny nitpick

File: game/src/sandbox/dashboards/travel_times.rs [hunk 4]
```diff
@@ -483,19 +493,60 @@ fn contingency_table(ctx: &mut EventCtx, app: &App, filter: &Filter) -> Widget {
...
+    }
+    y_axis_labels.autocrop_dims = true;
+    y_axis_labels = y_axis_labels.autocrop();
+
+    batch.extend(Color::BLACK, bar_outlines);
 
     Widget::col(vec![
         Text::from_multiline(vec![
-            Line("Aggregate difference by trip length").small_heading(),
+            Line("Aggregate difference by trip duration").small_heading(),
```
Comment (dabreegster) [inline game/src/sandbox/dashboards/travel_times.rs:547]: Good catch ^^;

File: geom/src/duration.rs (hunk 1 above)
Comment (dabreegster) [reply geom/src/duration.rs:88]: Eep -- how did this get by for so long? Maybe I meant "deciseconds". Looking at the call sites, winding up with centiseconds looks fine.

File: geom/src/duration.rs (hunk 2 above)
Comment (dabreegster) [reply geom/src/duration.rs:218]: This is pretty nice! And I forgot all about doctests, perfect way to describe the behavior.

File: geom/src/duration.rs [hunk 5]
```diff
@@ -196,6 +202,50 @@ impl Duration {
...
+        let (hours, minutes, seconds, remainder) = self.get_parts();
+        if hours == 0 && minutes == 0 && seconds == 0 && remainder == 0 {
+            return "0".to_string();
+        }
+
+        let sign = if self < Duration::ZERO { "-" } else { "" };
+
+        let (whole, part, unit) = {
```
Comment (dabreegster) [inline geom/src/duration.rs:226]: Elegant implementation!

**Review by dabreegster — APPROVED** [review]

File: game/src/sandbox/dashboards/travel_times.rs (hunk 3 above)
Comment (dabreegster) [reply game/src/sandbox/dashboards/travel_times.rs:529]: I'll fix this in a followup commit, which I need anyway to do the whole `[rebuild] [release]` dance for github actions

**Comment by dabreegster** [comment]
Merging, then will start a release

File: geom/src/duration.rs (hunk 1 above)
Comment (dabreegster) [reply geom/src/duration.rs:88]: I was wrong -- we do only want to print one decimal place. Loading screens with two decimal places are kind of overwhelming. :)   Fixed in 7399f2aff2ed0f61f8202bfadc721df5078e7583

**Pull request merged**